src/plugins/discover/public/application/apps/main/services/use_discover_state.ts

```diff
@@ -148,7 +150,7 @@ export function useDiscoverState({
   const resetSavedSearch = useCallback(
     async (id?: string) => {
       const newSavedSearch = await services.getSavedSearchById(id);
-      newSavedSearch.searchSource.setField('index', indexPattern);
+      newSavedSearch.searchSource.setField('index', savedSearchDefaultIp.current);
```

Comment:
this would be fine, but it would mean, that when clicking on `New` not the last index pattern selected would be chosen, which was considered as a bug recently: #103239 (But wasn't anymore in master)
 
So I think the best solution would be, if `resetSavedSearch` would get another argument depending on which the index pattern would be set to the one persisted in the saved search, or the last one selected in UI. With this I think the `useRef` would no longer be necessary, and both use cases would be covered

Reply:
You are right, fixed.